src/main/java/com/soboleoko/medicalclinic/PatientController.java

```diff
@@ -0,0 +1,56 @@
+package com.soboleoko.medicalclinic;
+
+import com.soboleoko.medicalclinic.model.Patient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/patients")
+public class PatientController {
+
+
+        private final PatientService patientService;
+
+        @Autowired
+        public PatientController(PatientService patientService) {
+            this.patientService = patientService;
+        }
+
+        @GetMapping
+        public List<Patient> getAllPatients() {
+            return patientService.getAllPatients();
+        }
+
+        @GetMapping("/{email}")
+        public ResponseEntity<Patient> getPatientByEmail(@PathVariable String email) {
```

Comment:
ResponseEntity to klasa reprezentująca odpowiedź HTTP. W sytuacji jak używasz adnotacji @RestController to automatycznie masz zamiane tego co zwracasz czyli np tak jak w powyzszym endpoincie zwracasz Patient to ten Patient trafia do body a Spring automatycznie obudowuje to body (Patient) w odpowiedź HTTP z statusem sukcesu 200 OK jeśli zapytanie się powiedzie. Defaultowy mechanizm Springa w sytuacji kiedy poleci w ramach przetwarzania zapytania wyjątek automatycznie zmapuje to na odpowiedź HTTP z kodem 500 INTERNAL_SERVER_ERROR.